pom.xml

```diff
@@ -525,20 +526,20 @@ under the License.
       <dependency>
         <groupId>org.hamcrest</groupId>
         <artifactId>hamcrest</artifactId>
-        <version>2.2</version>
+        <version>${hamcrestVersion}</version>
+        <scope>test</scope>
+      </dependency>
+      <dependency>
+        <groupId>org.hamcrest</groupId>
+        <artifactId>hamcrest-core</artifactId>
+        <version>${hamcrestVersion}</version>
         <scope>test</scope>
       </dependency>
       <dependency>
         <groupId>junit</groupId>
         <artifactId>junit</artifactId>
         <version>${junitVersion}</version>
         <scope>test</scope>
-        <exclusions>
```

Comment:
Shouldn't you actually keep the exclusion to not pull older version of hamcrest that conflicts with the one you defined on the top level?

Reply:
I recommend to look at https://hamcrest.org/JavaHamcrest/distributables#maven-upgrade-example